apps/docs/registry.mdx

```diff
@@ -0,0 +1,235 @@
+npx codemod publish
+```
+
+Learn more about the [`publish` command and available options](/cli/cli-reference#codemod-publish).
+
+During the first publish, you'll be prompted to:
+- Choose an organization or user scope
+- Set access level (public or private)
+- Confirm package metadata
+
+**Unpublishing packages:**
+
+```bash
+# Unpublish a package
+npx codemod unpublish <package-name>
+```
+
+<Warning>
+  Unpublishing is a destructive action. Consider deprecating packages instead of unpublishing if others may depend on them.
+</Warning>
+
+Learn more about the [`unpublish` command and available options](/cli/cli-reference#codemod-unpublish).
+
+### Configuring Access and Visibility
+
+In your `codemod.yaml`, configure registry settings:
+
+```yaml codemod.yaml
+registry:
+  access: "public"      # or "private", "pro"
+  visibility: "public"  # or "private", "org", "user"
+```
+
+**Access levels:**
+- `public` – Anyone can run the codemod
+- `private` – Only the owner can run the codemod
+- `pro` – Only Pro plan users can run the codemod
+
+**Visibility levels:**
+- `public` – Everyone can see the package in search and listings
+- `private` – Only the owner can see the package
+- `org` – Members of the package's organization scope can see it
+- `user` – Visible only to the publishing user
+
+<Tip>
+  Access and visibility work together. For example, `visibility: public` with `access: pro` shows the package publicly, but only Pro users can run it.
+</Tip>
+
+Learn more about [configuring packages](/cli/packages/building-workflows#package-metadata-codemodyaml).
+
+## Publishing from CI/CD
+
+The easiest way to set up CI/CD publishing is to use the [migrations template repository](https://github.com/codemod/migrations-template).
+
+<Card title="Migrations Template" icon="github" href="https://github.com/codemod/migrations-template">
+  The migrations template repository includes a preconfigured GitHub Action workflow for automatic publishing.
+</Card>
+
+This GitHub template includes:
+
+- Preconfigured GitHub Action workflow for automatic publishing
+- Setup instructions for GitHub App installation and API key configuration
+- Repository structure optimized for codemod packages
+
+After setup, codemods are automatically published to the registry when PRs are merged. See the template repository for complete setup instructions.
+
+## Using Packages in Workflows
+
+You can reference registry packages in your workflow definitions:
+
+```yaml workflow.yaml
+steps:
+  - name: Run registry codemod
+    codemod:
+      source: "@nodejs/create-require-from-path@1.1.0"
+```
+
+Learn more about [using Codemod packages in workflows](/cli/packages/building-workflows#codemod-registry-step).
+
+## Best Practices
+
+### Package Naming
+
+- Use descriptive names that indicate the codemod's purpose
+- Follow the naming convention: `/^[a-z0-9-_/]+$/`
+- Use scoped names for better discoverability (e.g., `@scope/package-name`)
```

Comment:
since folks can pick any available scope, I don’t think we should encourage using one without a reason. Instead, we should suggest reserving a scope by adding their GitHub org to our Codemod platform via installing the Codemod app on one of their codemod repos. Once the scope is reserved, only admins of that org can publish codemods with a name starting with that scope.